internal/cli/cmd/cluster/dataprotection.go

```diff
@@ -975,3 +1015,106 @@ func (o *editBackupPolicyOptions) applyChanges(backupPolicy *dpv1alpha1.BackupPo
+			return err
+		}
+		if err := o.printBackupObj(backupObj); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (o *describeBackupOptions) printBackupObj(obj *dpv1alpha1.Backup) error {
+	printer.PrintLineWithTabSeparator(
+		printer.NewPair("Name", obj.Name),
+		printer.NewPair("Cluster", obj.Status.SourceCluster),
+		printer.NewPair("Namespace", obj.Namespace),
+	)
+	printer.PrintLine("\nSpec:")
+	realPrintPairStringToLine("Type", string(obj.Spec.BackupType))
+	realPrintPairStringToLine("Policy Name", obj.Spec.BackupPolicyName)
+
+	printer.PrintLine("\nStatus:")
+	realPrintPairStringToLine("Phase", string(obj.Status.Phase))
+	realPrintPairStringToLine("Failure Reason", obj.Status.FailureReason)
+	realPrintPairStringToLine("Total Size", obj.Status.TotalSize)
+	realPrintPairStringToLine("Backup Tool", obj.Status.BackupToolName)
+	realPrintPairStringToLine("PVC Name", obj.Status.PersistentVolumeClaimName)
+	if obj.Status.AvailableReplicas != nil {
+		realPrintPairStringToLine("Available Replicas", string(*obj.Status.AvailableReplicas))
+	}
+	if obj.Status.Duration != nil {
+		realPrintPairStringToLine("Duration", duration.HumanDuration(obj.Status.Duration.Duration))
+	}
+	realPrintPairStringToLine("Expiration Time", util.TimeFormat(obj.Status.Expiration))
+	realPrintPairStringToLine("Start Time", util.TimeFormat(obj.Status.StartTimestamp))
+	realPrintPairStringToLine("Completion Time", util.TimeFormat(obj.Status.CompletionTimestamp))
+
+	if obj.Status.Manifests != nil {
+		printer.PrintLine("\nManifests:")
+		realPrintPairStringToLine("Target", obj.Status.Manifests.Target)
+		if obj.Status.Manifests.BackupLog != nil {
+			realPrintPairStringToLine("Log Start Time", util.TimeFormat(obj.Status.Manifests.BackupLog.StartTime))
+			realPrintPairStringToLine("Log Stop Time", util.TimeFormat(obj.Status.Manifests.BackupLog.StopTime))
+		}
+		if obj.Status.Manifests.BackupTool != nil {
+			realPrintPairStringToLine("File Path", obj.Status.Manifests.BackupTool.FilePath)
+			realPrintPairStringToLine("Volume Name", obj.Status.Manifests.BackupTool.VolumeName)
+			realPrintPairStringToLine("Upload Total Size", obj.Status.Manifests.BackupTool.UploadTotalSize)
+			realPrintPairStringToLine("Checksum", obj.Status.Manifests.BackupTool.Checksum)
+			realPrintPairStringToLine("Checkpoint", obj.Status.Manifests.BackupTool.Checkpoint)
+		}
+		if obj.Status.Manifests.Snapshot != nil {
+			realPrintPairStringToLine("Snapshot Name", obj.Status.Manifests.Snapshot.VolumeSnapshotName)
+			realPrintPairStringToLine("Snapshot Content Name", obj.Status.Manifests.Snapshot.VolumeSnapshotContentName)
+		}
+		for k, v := range obj.Status.Manifests.UserContext {
+			realPrintPairStringToLine(k, v)
+		}
+	}
+
+	// get all events about cluster
+	events, err := o.client.CoreV1().Events(o.namespace).Search(scheme.Scheme, obj)
+	if err != nil {
+		return err
+	}
+
+	// print the warning events
+	printer.PrintAllWarningEvents(events, o.Out)
```

Comment:
Is it necessary to output cluster event?

Reply:
Output cluster events is a bit beyond the scope of the command, I added an improvement to output pod logs when the backup fails.